chia/plotters/bladebit.py

```diff
@@ -0,0 +1,199 @@
+import asyncio
+import json
+import traceback
+import subprocess
+import os
+import sys
+import logging
+
+from pathlib import Path
+from typing import Any, Dict, Optional
+from chia.plotting.create_plots import resolve_plot_keys
+from chia.plotters.plotters_util import run_plotter, run_command
+
+log = logging.getLogger(__name__)
+
+
+BLADEBIT_PLOTTER_DIR = "bladebit"
+BLADEBIT_MIN_RAM_GIBS = 416
+
+
+def is_bladebit_supported() -> bool:
+    return sys.platform.startswith("linux") or sys.platform in ["win32", "cygwin"]
+
+
+def meets_memory_requirement(plotters_root_path: Path) -> bool:
+    have_enough_memory: bool = False
+    if get_bladebit_executable_path(plotters_root_path).exists():
+        try:
+            proc = subprocess.run(
+                [os.fspath(get_bladebit_executable_path(plotters_root_path)), "--memory-json"],
+                capture_output=True,
+                text=True,
+            )
+            memory_info: Dict[str, int] = json.loads(proc.stdout)
+            total_bytes: int = memory_info.get("total", -1)
+            required_bytes: int = memory_info.get("required", 0)
+            have_enough_memory = total_bytes >= required_bytes
+        except Exception as e:
+            print(f"Failed to determine bladebit memory requirements: {e}")
+
+    return have_enough_memory
+
+
+def get_bladebit_install_path(plotters_root_path: Path) -> Path:
+    return plotters_root_path / BLADEBIT_PLOTTER_DIR
+
+
+def get_bladebit_executable_path(plotters_root_path: Path) -> Path:
+    bladebit_exec: str = "bladebit"
+    if sys.platform in ["win32", "cygwin"]:
+        bladebit_exec = "bladebit.exe"
+    return get_bladebit_install_path(plotters_root_path) / ".bin/release" / bladebit_exec
+
+
+def get_bladebit_install_info(plotters_root_path: Path) -> Optional[Dict[str, Any]]:
+    info: Dict[str, Any] = {"display_name": "BladeBit Plotter"}
+    installed: bool = False
+    supported: bool = is_bladebit_supported()
+
+    if get_bladebit_executable_path(plotters_root_path).exists():
+        version: Optional[str] = None
+        try:
+            proc = subprocess.run(
+                [os.fspath(get_bladebit_executable_path(plotters_root_path)), "--version"],
+                capture_output=True,
+                text=True,
+            )
+            version = proc.stdout.strip()
+        except Exception as e:
+            print(f"Failed to determine bladebit version: {e}")
+
+        if version is not None:
+            installed = True
+            info["version"] = version
+        else:
+            installed = False
+
+    info["installed"] = installed
+    if installed is False:
+        info["can_install"] = supported
+
+    if supported and meets_memory_requirement(plotters_root_path) is False:
+        info["bladebit_memory_warning"] = f"BladeBit requires at least {BLADEBIT_MIN_RAM_GIBS} GiB of RAM to operate"
```

Comment:
BladeBit now outputs the minimum amount so we don't need to hardcode `BLADEBIT_MIN_RAM_GIBS`.

Reply:
Done